src/components/browse/components/VirtualHrefController.js

```diff
@@ -105,10 +106,41 @@ export class VirtualHrefController extends React.PureComponent {
 
         let nextHrefFull = null;
         let virtualCompoundFilterSet = null;
+
         if (typeof navigationTarget === "string") {
             // There is (very large) chance that `nextHref` does not have domain name, path, etc.
             // Resolve based on current virtualHref (else AJAX call may auto-resolve relative to browser URL).
-            nextHrefFull = url.resolve(currentHref || "/search/", navigationTarget);
+            nextHrefFull = url.resolve((currentHref || "/search/"), navigationTarget);
+
+            // Divide URL into parts and put into a virtualCompoundFilterSet, in effect making all virtual search
+            // requests into POST requests.
+            const targetHrefParts = url.parse(nextHrefFull, true);
+            const gParts = {};
+            const fParts = {};
+            Object.keys(targetHrefParts.query).forEach(function(k){
+                if (k === "type" || k === "sort" || k === "additional_facet") {
+                    gParts[k] = targetHrefParts.query[k];
+                } else {
+                    fParts[k] = targetHrefParts.query[k];
+                }
+            });
+            const gStr = queryString.stringify(gParts);
+            const fStr = queryString.stringify(fParts);
```

Comment:
Would probably be helpful for readability to give these variables better names. Is there a reason for "g" and "f"? I had to actually go in and log them to figure out what they were/couldn't tell from the name and context alone.

Reply:
Not too good of a reason - 'g' referred to what goes in ***g***lobal_flag and 'f' for ***f***ilter_blocks.query of the POST body. Will rename the vars.